src/cilint.ml

```diff
@@ -37,232 +29,126 @@ let m1 = minus_big_int unit_big_int
 let nobits (b:big_int) : bool =
   sign_big_int b = 0 || compare_big_int b m1 = 0
 
-let big_int_of_cilint (c:cilint) : big_int =
-  match c with
-  | Small i -> big_int_of_int i
-  | Big b -> b
-
-let cilint_of_big_int (b:big_int) : cilint =
-  if is_int_big_int b then
-    Small (int_of_big_int b)
-  else
-    Big b
-
-let neg_cilint c =
-  match c with
-  | Small i when i <> min_int -> Small (-i)
-  | _ -> Big (minus_big_int (big_int_of_cilint c))
-
-(* Apply big_int 'op' to two cilints, returning a cilint *)
-let b op c1 c2 = cilint_of_big_int (op (big_int_of_cilint c1) (big_int_of_cilint c2))
+let big_int_of_cilint (c:cilint) : big_int = c
+let cilint_of_big_int (b:big_int) : cilint = b
 
-let add_cilint = b add_big_int
-let sub_cilint = b sub_big_int
-let mul_cilint = b mult_big_int
-let div_cilint = b div_big_int
-let mod_cilint = b mod_big_int
+let neg_cilint = minus_big_int
 
-let compare_cilint (c1:cilint) (c2:cilint) : int =
-  match c1, c2 with
-  | Small i1, Small i2 -> compare i1 i2
-  | _ -> compare_big_int (big_int_of_cilint c1) (big_int_of_cilint c2)
+let add_cilint = add_big_int
+let sub_cilint = sub_big_int
+let mul_cilint = mult_big_int
+let div_cilint = div_big_int
+let mod_cilint = mod_big_int
 
-let is_zero_cilint (c:cilint) : bool =
-  match c with
-  | Small i -> i = 0
-  | Big b -> sign_big_int b = 0
+let compare_cilint = compare_big_int
+let is_zero_cilint (c:cilint) : bool = sign_big_int c = 0
 
-let negative_cilint (c:cilint) : bool =
-  match c with
-  | Small i -> i < 0
-  | Big b -> sign_big_int b < 0
+let negative_cilint (c:cilint) : bool = sign_big_int c < 0
 
-let cilint_of_int (i:int) : cilint = Small i
+let cilint_of_int = big_int_of_int
+let int_of_cilint = int_of_big_int
 
-let int_of_cilint (c:cilint) : int =
-  match c with
-  | Small i -> i
-  | Big b -> int_of_big_int b
-
-let cilint_of_int64 (i64:int64) : cilint =
-  if Int64.compare i64 (Int64.of_int min_int) >= 0 &&
-    Int64.compare i64 (Int64.of_int max_int) <= 0 then
-    Small (Int64.to_int i64)
-  else
-    (* We convert 30 bits at a time *)
-    let rec loop i mul acc =
-      if i = 0L then acc
-      else if i = -1L then sub_big_int acc mul
-      else
-	let lo30 = Int64.to_int (Int64.logand i 0x3fffffffL) in
-	loop (Int64.shift_right i 30) (mult_big_int mul b30)
-	  (add_big_int acc (mult_big_int mul (big_int_of_int lo30)))
-    in Big (loop i64 unit_big_int zero_big_int)
```

Comment:
Do you know why this was done so? Maybe it should stay for whatever reason, if it needed to be that way even with the big ints before.

Reply:
https://github.com/goblint/cil/blob/a3c91aa6e8f946fec9a9a13361b051a73b12a65c/src/cilint.ml#L20-L22

I assume it was to avoid the bug they mentioned here.